src/GSoft.Extensions.MediatR.ApplicationInsights/RequestApplicationInsightsBehavior.cs

```diff
@@ -21,22 +22,38 @@ public Task<TResponse> Handle(TRequest request, RequestHandlerDelegate<TResponse
 
     private async Task<TResponse> HandleWithTelemetry(TRequest request, RequestHandlerDelegate<TResponse> next)
     {
-        using (var operation = this._telemetryClient.StartOperation<DependencyTelemetry>(request.GetType().Name))
+        var operation = this._telemetryClient.StartOperation<DependencyTelemetry>(request.GetType().Name);
+
+        // Originating activity must be captured AFTER that the operation is created
+        // Because ApplicationInsights SDK creates another intermediate Activity
+        var originatingActivity = Activity.Current;
+
+        try
         {
             operation.Telemetry.Type = ApplicationInsightsConstants.TelemetryType;
+            var result = await next().ConfigureAwait(false);
+            operation.Telemetry.Success = true;
 
-            try
-            {
-                var result = await next().ConfigureAwait(false);
-                operation.Telemetry.Success = true;
+            return result;
+        }
+        catch (Exception ex)
+        {
+            operation.Telemetry.Success = false;
+            operation.Telemetry.Properties.TryAdd(ApplicationInsightsConstants.Exception, ex.ToString());
 
-                return result;
+            throw;
+        }
+        finally
+        {
+            // The dependency telemetry is sent when the operation is disposed
+            if (originatingActivity == null)
```

Comment:
Can this be null? (this code path don't seems to be covered in the tests)

Reply:
Yes `Activity.Current` can be return null if there is no current activity being observed.

If we want to test this path we need to duplicate the current test class and fixture, then remove InMemoryActivityTracker from the fixture so we don't observe Activities.